File: backoffice/src/mocks/landing.ts
import type { CrudItemType } from "@/types/CrudItem";

export type LandingSection = "Hero" | "Favoritos" | "Depoimentos" | "CTA Final";
export type LandingStatus = "Ativo" | "Inativo";

export interface MockLandingContentItem extends CrudItemType {
  secao: LandingSection;
  titulo: string;
  subtitulo: string;
  descricao: string;
  imagem: string;
  botaoTexto: string;
  botaoLink: string;
  status: LandingStatus;
}

export const MOCK_LANDING_CONTENT: MockLandingContentItem[] = [
  {
    id: "1",
    name: "Hero - Cookies que fazem sorrir",
    secao: "Hero",
    titulo: "Cookies que fazem sorrir",
    subtitulo: "Cookies & Coffee Break",
    descricao:
      "Feitos à mão com ingredientes de verdade, muito amor e uma pitada de magia. Cada mordida é um abraço quentinho.",
    imagem: "cookie-home.png",
    botaoTexto: "Ver cookies",
    botaoLink: "/shopping",
    status: "Ativo",
  },
  {
    id: "2",
    name: "Favoritos - Cookie Red",
    secao: "Favoritos",
    titulo: "Cookie Red",
    subtitulo: "Os mais amados!",
    descricao: "Cookie artesanal com chocolate branco e notas suaves de baunilha.",
    imagem: "cookie-red.png",
    botaoTexto: "Quero Esse!",
    botaoLink: "/shopping",
    status: "Ativo",
  },
  {
    id: "3",
    name: "Favoritos - Chocolate Ao Leite",
    secao: "Favoritos",
    titulo: "Chocolate Ao Leite",
    subtitulo: "Os mais amados!",
    descricao: "Cookie artesanal com chocolate belga ao leite, crocante por fora e derretido por dentro.",
    imagem: "cookie-choco-chunk.png",
    botaoTexto: "Quero Esse!",
    botaoLink: "/shopping",
    status: "Ativo",
  },
  {
    id: "4",
    name: "Favoritos - Double Chocolate",
    secao: "Favoritos",
    titulo: "Double Chocolate",
    subtitulo: "Os mais amados!",
    descricao: "Cookie artesanal com massa de cacau 70% e gotas de chocolate meio amargo.",
    imagem: "cookie-double-choc.png",
    botaoTexto: "Quero Esse!",
    botaoLink: "/shopping",
    status: "Ativo",
  },
  {
    id: "5",
    name: "Depoimentos - Ana Clara",
    secao: "Depoimentos",
    titulo: "Ana Clara",
    subtitulo: "Declarações de Amor",
    descricao:
      "Gente, eu CHOREI comendo o de chocolate ao leite. Não é exagero. É viciante demais!",
    imagem: "",
    botaoTexto: "",
    botaoLink: "",
    status: "Ativo",
  },
  {
    id: "6",
    name: "Depoimentos - Mariana",
    secao: "Depoimentos",
    titulo: "Mariana",
    subtitulo: "Declarações de Amor",
    descricao:
      "Comprei para dividir e me arrependi. Queria tudo pra mim. A massa é macia e o recheio é perfeito.",
    imagem: "",
    botaoTexto: "",
    botaoLink: "",
    status: "Ativo",
  },
  {
    id: "7",
    name: "Depoimentos - Carlos",
    secao: "Depoimentos",
    titulo: "Carlos",
    subtitulo: "Declarações de Amor",
    descricao:
      "Pedi no fim da tarde e chegou quentinho. Virou meu ritual de sexta com café.",
    imagem: "",
    botaoTexto: "",
    botaoLink: "",
    status: "Ativo",
  },
  {
    id: "8",
    name: "CTA Final - Fazer meu pedido",
    secao: "CTA Final",
    titulo: "Tá esperando o que pra experimentar?",
    subtitulo: "Peça online e receba seus cookies quentinhos em minutos.",
    descricao:
      "Delivery ou retirada — você escolhe!",
    imagem: "",
    botaoTexto: "Fazer meu pedido 🍪",
    botaoLink: "/cart",
    status: "Ativo",
  },
  {
    id: "9",
    name: "CTA Final - Ver cardápio",
    secao: "CTA Final",
    titulo: "Tá esperando o que pra experimentar?",
    subtitulo: "Peça online e receba seus cookies quentinhos em minutos.",
    descricao:
      "Delivery ou retirada — você escolhe!",
    imagem: "",
    botaoTexto: "Ver cardápio",
    botaoLink: "/shopping",
    status: "Ativo",
  },
];

File: backend/internal/services/landing.go
package services

import (
	"backend/internal/models"
	"backend/internal/repository"
)

type LandingService interface {
	GetAll() ([]models.LandingContent, error)
	GetActive() ([]models.LandingContent, error)
	Create(input models.LandingContentInput) (*models.LandingContent, error)
	Update(id uint, input models.LandingContentInput) (*models.LandingContent, error)
	Delete(id uint) error
	InitializeDefaults() error
}

type landingService struct{ repo repository.LandingRepository }

func NewLandingService(repo repository.LandingRepository) LandingService {
	return &landingService{repo: repo}
}

func (s *landingService) GetAll() ([]models.LandingContent, error) {
	return s.repo.GetAll()
}

func (s *landingService) GetActive() ([]models.LandingContent, error) {
	return s.repo.GetActive()
}

func (s *landingService) Create(input models.LandingContentInput) (*models.LandingContent, error) {
	content := landingFromInput(input)
	if err := s.repo.Create(content); err != nil {
		return nil, err
	}
	return content, nil
}

func (s *landingService) Update(id uint, input models.LandingContentInput) (*models.LandingContent, error) {
	content, err := s.repo.GetByID(id)
	if err != nil {
		return nil, err
	}
	content.Name = input.Name
	content.Secao = input.Secao
	content.Titulo = input.Titulo
	content.Subtitulo = input.Subtitulo
	content.Descricao = input.Descricao
	content.Imagem = input.Imagem
	content.BotaoTexto = input.BotaoTexto
	content.BotaoLink = input.BotaoLink
	content.Status = landingStatus(input.Status)
	if err := s.repo.Update(content); err != nil {
		return nil, err
	}
	return content, nil
}

func (s *landingService) Delete(id uint) error {
	return s.repo.Delete(id)
}

func (s *landingService) InitializeDefaults() error {
	total, err := s.repo.Count()
	if err != nil {
		return err
	}
	if total > 0 {
		return nil
	}
	for _, input := range defaultLandingContent {
		if _, err := s.Create(input); err != nil {
			return err
		}
	}
	return nil
}

func landingFromInput(input models.LandingContentInput) *models.LandingContent {
	return &models.LandingContent{
		Name:       input.Name,
		Secao:      input.Secao,
		Titulo:     input.Titulo,
		Subtitulo:  input.Subtitulo,
		Descricao:  input.Descricao,
		Imagem:     input.Imagem,
		BotaoTexto: input.BotaoTexto,
		BotaoLink:  input.BotaoLink,
		Status:     landingStatus(input.Status),
	}
}

func landingStatus(status string) string {
	if status == "Inativo" {
		return "Inativo"
	}
	return "Ativo"
}

var defaultLandingContent = []models.LandingContentInput{
	{
		Name:       "Hero - Cookies que fazem sorrir",
		Secao:      "Hero",
		Titulo:     "Cookies que fazem sorrir",
		Subtitulo:  "Cookies & Coffee Break",
		Descricao:  "Feitos à mão com ingredientes de verdade, muito amor e uma pitada de magia. Cada mordida é um abraço quentinho.",
		Imagem:     "cookie-home.png",
		BotaoTexto: "Ver cookies",
		BotaoLink:  "/shopping",
		Status:     "Ativo",
	},
	{
		Name:       "Favoritos - Cookie de morango",
		Secao:      "Favoritos",
		Titulo:     "Cookie de morango",
		Subtitulo:  "Os mais amados!",
		Descricao:  "Cookie artesanal com pedaços de morango e ganache de chocolate",
		Imagem:     "cookie-morango.jpg",
		BotaoTexto: "Quero Esse!",
		BotaoLink:  "/shopping",
		Status:     "Ativo",
	},
	{
		Name:       "Favoritos - Cookie de Matcha",
		Secao:      "Favoritos",
		Titulo:     "Cookie de Matcha",
		Subtitulo:  "Os mais amados!",
		Descricao:  "Cookie artesanal com pedaços de Matcha e ganache de chocolate",
		Imagem:     "cookie-matcha.jpg",
		BotaoTexto: "Quero Esse!",
		BotaoLink:  "/shopping",
		Status:     "Ativo",
	},
	{
		Name:       "Favoritos - Cookie de chocolate",
		Secao:      "Favoritos",
		Titulo:     "Cookie de chocolate",
		Subtitulo:  "Os mais amados!",
		Descricao:  "Cookie artesanal com pedaços de chocolate e ganache de chocolate",
		Imagem:     "cookie-choco-chunk.jpg",
		BotaoTexto: "Quero Esse!",
		BotaoLink:  "/shopping",
		Status:     "Ativo",
	},
	{
		Name:      "Depoimentos - Ana Clara",
		Secao:     "Depoimentos",
		Titulo:    "Ana Clara",
		Subtitulo: "Declarações de Amor",
		Descricao: "Gente, eu CHOREI comendo o de caramelo salgado. Não é exagero. É viciante demais!",
		Status:    "Ativo",
	},
	{
		Name:      "Depoimentos - Mariana",
		Secao:     "Depoimentos",
		Titulo:    "Mariana",
		Subtitulo: "Declarações de Amor",
		Descricao: "Comprei para dividir e me arrependi. Queria tudo pra mim. A massa é macia e o recheio é perfeito.",
		Status:    "Ativo",
	},
	{
		Name:      "Depoimentos - Carlos",
		Secao:     "Depoimentos",
		Titulo:    "Carlos",
		Subtitulo: "Declarações de Amor",
		Descricao: "Pedi no fim da tarde e chegou quentinho. Virou meu ritual de sexta com café.",
		Status:    "Ativo",
	},
	{
		Name:       "CTA Final - Fazer meu pedido",
		Secao:      "CTA Final",
		Titulo:     "Tá esperando o que pra experimentar?",
		Subtitulo:  "Peça online e receba seus cookies quentinhos em minutos.",
		Descricao:  "Delivery ou retirada — você escolhe!",
		BotaoTexto: "Fazer meu pedido",
		BotaoLink:  "/cart",
		Status:     "Ativo",
	},
	{
		Name:       "CTA Final - Ver cardápio",
		Secao:      "CTA Final",
		Titulo:     "Tá esperando o que pra experimentar?",
		Subtitulo:  "Peça online e receba seus cookies quentinhos em minutos.",
		Descricao:  "Delivery ou retirada — você escolhe!",
		BotaoTexto: "Ver cardápio",
		BotaoLink:  "/shopping",
		Status:     "Ativo",
	},
}
